truss/templates/trtllm/model/model.py

```diff
@@ -116,14 +118,19 @@ async def predict(self, model_input):
         model_input = ModelInput(**model_input)
         result_iterator = self.triton_client.infer(model_input)
 
-        async def generate():
+        async def generate() -> AsyncIterator[InferenceResult]:
             async for result in result_iterator:
-                yield result
+                yield result.value
 
         if model_input.stream:
             return generate()
-        else:
-            if self.uses_openai_api:
-                return "".join(generate())
-            else:
-                return {"text": "".join(generate())}
+
+        response = ""
+        async for value in generate():
+            if value.status_code != 200:
```

Comment:
Instead of comparing against a literal (`200`), why not make this an API of `InferenceResult`? E.g. add a method `get_result_or_raise()` that returns the prediction in the success case and raises this exception in the error case?

This proposal would encapsulate the internal details of `InferenceResult` in a way that it can safely and easily be used externally, so the call site doesn't need to know about the semantics of `status_codde == 200`.